### PR TITLE
Measure Graphql complexity

### PR DESCRIPTION
**Pull request by shemnon**
Measure the complexity of GraphQL queries

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

### CODE REVIEW

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLProvider.java [hunk 1]
```diff
@@ -31,14 +33,20 @@
 
 public class GraphQLProvider {
 
+  public static final int MAX_COMPLEXITY = 200;
```
Comment (atoulme) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLProvider.java:36]: Is this to be configurable?

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLProvider.java [hunk 2]
```diff
@@ -74,4 +82,18 @@ private static RuntimeWiring buildWiring(final GraphQLDataFetchers graphQLDataFe
...
+      final FieldComplexityEnvironment environment, final int childComplexity) {
+    final String childTypeName = environment.getParentType().getName();
+    final String fieldName = environment.getField().getName();
+
+    if (childTypeName.equals("Transaction") && fieldName.equals("block")) {
+      return childComplexity + 100;
+    } else if (childTypeName.equals("__Type") && fieldName.equals("fields")) {
+      return childComplexity + 100;
+    } else {
+      return childComplexity + 1;
```
Comment (atoulme) [inline ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLProvider.java:96]: Is there a published algorithm for this?

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/graphql/EthGraphQLHttpBySpecErrorCaseTest.java [hunk 3]
```diff
@@ -1,88 +0,0 @@
...
-import okhttp3.RequestBody;
-import okhttp3.Response;
-import org.assertj.core.api.Assertions;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-import org.junit.runners.Parameterized.Parameters;
-
-@RunWith(Parameterized.class)
-public class EthGraphQLHttpBySpecErrorCaseTest extends AbstractEthGraphQLHttpServiceTest {
```
Comment (atoulme) [inline ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/graphql/EthGraphQLHttpBySpecErrorCaseTest.java:35]: Why is that test removed?

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLProvider.java (hunk 1 above)
Comment (shemnon) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLProvider.java:36]: No.  Normal queries won't break that limit.

File: ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLProvider.java (hunk 2 above)
Comment (shemnon) [reply ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLProvider.java:96]: This is a custom algorithm, focused on our schema.

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/graphql/EthGraphQLHttpBySpecErrorCaseTest.java (hunk 3 above)
Comment (shemnon) [reply ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/graphql/EthGraphQLHttpBySpecErrorCaseTest.java:35]: There were no test being loaded up in the parameters section.  All tests (error or non error) are all in `EthGraphQLHttpBySpecTest`

**Pull request merged**